docs/kernel_finder.rst

```diff
@@ -0,0 +1,183 @@
+  instances of class :class:`KernelSpecProvider <jupyter_kernel_mgmt.discovery.KernelSpecProvider>` the text
+  ``{connection_file}`` in any argument will be replaced with the path to the
+  connection file.  However, subclasses of :class:`KernelSpecProvider <jupyter_kernel_mgmt.discovery.KernelSpecProvider>`
+  may choose to provide different substitutions, especially if they don't use a connection file.
+- **interrupt_mode** (optional): May be either ``signal`` or ``message`` and
+  specifies how a client is supposed to interrupt cell execution on this kernel,
+  either by sending an interrupt ``signal`` via the operating system's
+  signalling facilities (e.g. `SIGINT` on POSIX systems), or by sending an
+  ``interrupt_request`` message on the control channel (see
+  :ref:`kernel interrupt (FIXME ref jupyter_protocol) <jupyter_client:msging_interrupt>`).
+  If this is not specified
+  the client will default to ``signal`` mode.  Because providers are responsible
+  for interrupting the kernel they launch, interpretation of this field is purely
+  the responsibility of the provider.
+- **env** (optional): A dictionary of environment variables to set for the kernel.
+  These will be added to the current environment variables before the kernel is
+  started.
+- **metadata** (optional): A dictionary of additional attributes about this
+  kernel. Metadata added here should be namespaced for the tool reading and
+  writing that metadata.
+
+
+Launching kernels
+=================
+
+Launching kernels works similarly to their discovery.  To launch a previously discovered kernel,
+the kernel's `fully qualified kernel type` is provided to KernelFinder's
+:meth:`launch() <jupyter_kernel_mgmt.discovery.KernelFinder.launch>` method.
+
+.. note::
+   A **fully qualified kernel type** includes a prefix of the kernel's :ref:`provider id <provider_id>` followed by a
+   forward slash ('/').  For example, the ``python3`` kernel as provided by the ``KernelSpecProvider``
+   would have a fully qualified kernel type of ``spec/python3``.
+
+   If no prefix is found, ``KernelFinder`` will apply a prefix of ``spec/`` since nearly all existing
+   kernel specifications can be handled by the
+   :class:`KernelSpecProvider <jupyter_kernel_mgmt.discovery.KernelSpecProvider>`.
+
+KernelFinder's launch method then locates the provider and call's the specific kernel provider's
+:meth:`launch() <jupyter_kernel_mgmt.discovery.KernelProviderBase.launch>` method.
+
+:py:meth:`KernelFinder.launch(name, cwd=None, launch_params=None) <jupyter_kernel_mgmt.discovery.KernelProviderBase.launch>`
+takes two additional (and optional) arguments.
+
+**cwd** (optional) specifies the current working directory relative to the notebook that will be associated
+with the launched kernel.  For :class:`KernelSpecProvider-based <.KernelSpecProvider>` kernels, the kernel
+process will use this value as the working directory for the subsequent Popen subprocess.
```

Comment:
Should we mention that it's up to the kernel provider whether and how it uses the cwd value? It's not guaranteed that a path makes sense in the context where the kernel runs - it might not be able to see the notebook file, for instance.